src/http_microhttpd.c

```diff
@@ -589,12 +593,20 @@ static int encode_and_redirect_to_splashpage(struct MHD_Connection *connection,
 		}
 	}
 
-	if (encoded[0]) {
+	if (config->fas_port) {
+		target = (config->fas_remoteip ? config->fas_remoteip : config->gw_address);
+
+		// append clientip or authaction url
+		if (config->fas_secure_enabled) {
+			safe_asprintf(&splashpageurl, "http://%s:%u/%s?clientip=%s&redir=%s",
+				target, config->fas_port, config->fas_path, token, encoded);
+		} else {
+			safe_asprintf(&splashpageurl, "http://%s:%u/%s?authaction=http://%s:%u/%s/&tok=%s&redir=%s",
```

Comment:
haven't we used authaction already for some other purpose? Or was that only an internal variable?

Reply:
It is the exact same $authaction passed as a variable by mhd into splash.html when fas is disabled.
Fas must pass this back to nds after it has done its authentication so nds can change the client state from preauth to auth and call binauth if enabled.

Reply:
.....so nds can call binauth if enabled and change the client state from preauth to auth.